src/openrct2/localisation/LocalisationService.cpp

```diff
@@ -99,48 +95,59 @@ void LocalisationService::OpenLanguage(int32_t id)
     }
 
     std::string filename;
-    if (id != LANGUAGE_ENGLISH_UK)
-    {
-        filename = GetLanguagePath(LANGUAGE_ENGLISH_UK);
-        _languageFallback = LanguagePackFactory::FromFile(LANGUAGE_ENGLISH_UK, filename.c_str());
-    }
-
     filename = GetLanguagePath(id);
-    _languageCurrent = LanguagePackFactory::FromFile(id, filename.c_str());
-    if (_languageCurrent != nullptr)
+    auto preferredLanguage = LanguagePackFactory::FromFile(id, filename.c_str());
+    if (preferredLanguage != nullptr)
     {
         _currentLanguage = id;
+        _loadedLanguages.emplace_back(std::move(preferredLanguage));
         TryLoadFonts(*this);
     }
     else
     {
         throw std::runtime_error("Unable to open language " + std::to_string(id));
     }
+
+    const auto fallback = LanguagesDescriptors[id].fallback;
+    if (fallback != LANGUAGE_UNDEFINED)
+    {
+        filename = GetLanguagePath(fallback);
+        _loadedLanguages.emplace_back(LanguagePackFactory::FromFile(fallback, filename.c_str()));
```

Comment:
Doesn't this need to check for the return of `FromFile` to be different from `nullptr`? Since that's what the main language loading does?

Also, should `TryLoadFonts()` be retriggered?

Reply:
I doubt `TryLoadFonts()` would have to be triggered unless there was a fallback against Arabic or a CJK language from a language that doesn’t use the same writing system.

Reply:
Ok, but the question for checking the return of `FromFile` still applies here and below.